htdocs/product/card.php

```diff
@@ -1083,10 +1085,49 @@
 
 		// Batch number management
 		if (!empty($conf->productbatch->enabled)) {
-			print '<tr><td>'.$langs->trans("ManageLotSerial").'</td><td colspan="3">';
+			print '<tr><td>'.$langs->trans("ManageLotSerial").'</td><td>';
 			$statutarray = array('0' => $langs->trans("ProductStatusNotOnBatch"), '1' => $langs->trans("ProductStatusOnBatch"), '2' => $langs->trans("ProductStatusOnSerial"));
 			print $form->selectarray('status_batch', $statutarray, GETPOST('status_batch'));
-			print '</td></tr>';
+			print '</td>';
+			// Product specific batch number management
+			$status_batch = GETPOST('status_batch');
+			if ($status_batch !== '0') {
+				$tooltip = $langs->trans("GenericMaskCodes", $langs->transnoentities("Batch"), $langs->transnoentities("Batch"));
+				$tooltip .= $langs->trans("GenericMaskCodes2");
+				$tooltip .= $langs->trans("GenericMaskCodes3");
+				$tooltip .= $langs->trans("GenericMaskCodes4a", $langs->transnoentities("Batch"), $langs->transnoentities("Batch"));
+				$tooltip .= $langs->trans("GenericMaskCodes5");
+				print '<td id="mask_option">'.$langs->trans("ManageLotMask").'</td>';
+				if (($conf->global->PRODUCTBATCH_LOT_USE_PRODUCT_MASKS && $conf->global->PRODUCTBATCH_LOT_ADDON == 'mod_lot_advanced') || ($conf->global->PRODUCTBATCH_SN_USE_PRODUCT_MASKS && $conf->global->PRODUCTBATCH_SN_ADDON == 'mod_sn_advanced')) {
+					$inherited_mask_lot = $conf->global->LOT_ADVANCED_MASK;
+					$inherited_mask_sn = $conf->global->SN_ADVANCED_MASK;
+					print '<td id="field_mask">';
+					print $form->textwithpicto('<input type="text" class="flat" size="24" name="batch_mask" id="batch_mask_input">', $tooltip, 1, 1);
+					print '</td>';
+					print '<script type="text/javascript">
+								$(document).ready(function() {
+									$("#field_mask, #mask_option").addClass("hideobject");
+									$("#status_batch").on("change", function () {
+										var optionSelected = $("option:selected", this);
+										var valueSelected = this.value;
+										$("#field_mask, #mask_option").addClass("hideobject");
+										if (this.value == 1) {
+											$("#field_mask, #mask_option").toggleClass("hideobject");
+											$("#batch_mask_input").val("'.$inherited_mask_lot.'");
+										}
+										if (this.value == 2) {
+											$("#field_mask, #mask_option").toggleClass("hideobject");
+											$("#batch_mask_input").val("'.$inherited_mask_sn.'");
+										}
+									})
+								})
+							</script>';
+				} else {
+					print '<td colspan="2"></td>';
+				}
+			}
```

Comment:
the print '<td colspan="2"></td>'; should also be into this other else.

Reply:
I am not sure what you mean. What print am I missing ? I actually have put two "else", for add and edit.

Reply:
@altairis-noe he is talking about the else of "if ($status_batch !== '0') {" since you removed the colspan above.

Reply:
thanks @altairis-tof, I think I have it now !